snuba/datasets/cdc/groupassignee_entity.py

```diff
@@ -31,6 +31,8 @@ def __init__(self) -> None:
             abstract_column_set=schema.get_columns(),
             join_relationships={},
             writable_storage=storage,
+            required_filter_columns=None,
+            required_time_column=None,
```

Comment:
This is probably a distant concern but I guess it's at least theoretically possible for future conditions we may want to require on an entity to not fit neatly into one of these two categories. Maybe we want to allow GT/LT on something other than a time column, or somehow customise the project, org and time filters for an entity. I don't have any realistic use case right now since our data is essentially all time series by org or project, for which this works fine. Still I'm curious if this was a consideration when deciding on this format. Did you consider having the entity provide the matchers so they can be fully customised?

Reply:
Yes, that's why I moved the validation function into the Entity base class, and that function takes the entire query as an argument. Now if certain entities want custom validation, they can override the validation on the base class to do whatever custom work they want.

This actually came up as part of the sessions work, since we talked about having different required columns based on the granularity requested in the query. So the Sessions entity could override the validation, and have its own logic based on the granularity.